app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/general.ui.test.js

```diff
@@ -0,0 +1,347 @@
+import nock from 'nock';
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import content from '../manifest.json';
+import { solutionsApiUrl } from '../../../../../../../../config';
+import { nockAndErrorCheck, setState, authTokenInSession } from '../../../../../../../../test-utils/uiTestHelper';
+import { sessionKeys } from '../../../../../../../../helpers/routes/sessionHelper';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const pageUrl = 'http://localhost:1234/order/organisation/order-id/catalogue-solutions/neworderitem';
+
+const selectedPrice = {
+  priceId: 2,
+  provisioningType: 'ondemand',
+  type: 'flat',
+  currencyCode: 'GBP',
+  itemUnit: {
+    name: 'patient',
+    description: 'per patient',
+  },
+  timeUnit: {
+    name: 'month',
+    description: 'per month',
+  },
```

Comment:
On-demand prices don't have a time unit so we can delete the `timeUnit` property.